prov/efa/src/efa_domain.c

```diff
@@ -251,7 +245,12 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 		return -FI_EINVAL;
 	}
 
-	if (efa_mr_cache_enable) {
+	/*
+	 * If FI_MR_LOCAL is set, we do not want to use the
+	 * MR cache
+	 */
+	if (efa_mr_cache_enable && info->domain_attr && info->domain_attr->mr_mode & ~FI_MR_LOCAL) {
```

Comment:
This needs to happen before the fork support check, see line 237 in this file

Reply:
Wait why does it need to happen before? It won't be malloc'd before fork support so the cache won't be initialized, no?

Reply:
`efa_mr_cache_enable` will be 1 (the default) and NCCL will turn on fork support so the earlier check will fail. We'll never get to this if block.

Reply:
hmm alright